massgov/mayflower
# DP-20381 text overwrap

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)


## Description

Fix a left floated figure component to cover list style elements (bullets and numbers) in lists placed after the component in rich text.

The issue only occurs with left float.  The issue is not fixable by adjusting left side spacing of those lists since the floated element's right margin defines their positioning. Solving the issue by adding extra right margin to left float figure components without affecting the wrapping of following elements.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-20381)
- [Github issue]()

## Steps to Test


1. Create a sample with any content type with a rich text field.
2. Add an image, iframe or Tableau in the rich text field, set it align left.
3. Add a list after above element.
4. See the rendered page with those elements,
5. With the screen widths over 621px, find the floated element has the inline right margin.
6. With the screen widths less than 620px, it has no inline style.

Note:  When the field has multiple figure components consecutively, they don't have inline right margin even with the screen widths over 621px.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

**After**

<img width="909" alt="unorderedlist-after" src="https://user-images.githubusercontent.com/9633303/104637220-80acc980-5672-11eb-8cb6-b60b834081dd.png">

<img width="913" alt="orderedlist-after" src="https://user-images.githubusercontent.com/9633303/104637227-84405080-5672-11eb-9c05-8302cdb28839.png">

**Before**

<img width="913" alt="unordered-before" src="https://user-images.githubusercontent.com/9633303/104637257-8f937c00-5672-11eb-8c1f-264e0bfed0b4.png">

<img width="914" alt="ordered-before" src="https://user-images.githubusercontent.com/9633303/104637268-928e6c80-5672-11eb-93fd-bbc56723d517.png">


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


*

#### @TODO


*

#### Today I learned...
